### PR TITLE
Add allow.conflicting.dates settings in convert.inputs

### PR DESCRIPTION
**Pull request by tonygardella**
Allow there to be conflicting dates in dbfile.input.insert if exact.dates are not needed. 

## Description
There was an error where valid inputs with different dates were found but triggered an error because they did not have the correct file record associated. This will allow convert.input to create the correct input record regardless of date with the correct site, format, and machine and associate the correct dbfile record with it. 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue) #1197 
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by mdietze** [comment]
The issue Istem encountered of dates of existing met on other machines, and the fix of allowing conflicting dates, both need more thought.  Specifically, the whole point of the Inputs-DBfiles separation is to allow the same files to exist across multiple machines. This failure suggests that at least part of the problem comes from met.process not fully accounting for this. A quick fix that leads to redundant Inputs isn't the solution we're looking for.

**Comment by tonygardella** [comment]
That makes sense. I took a look at it and I think this can be handled in convert.inputs. Instead of saying  there isn't a valid input when there is not a local file associated with that input, I'll have it get that input id and then dbfile.insert the new file and associate it with that input.

**Comment by tonygardella** [comment]
latest change allows convert.input to find the correct input.record even if associated files are on a different machine. Next I need to sort out the download part, where I'll download all dates within that input record, and insert the path into the existing input record.

File: utils/R/convert.input.R [hunk 1]
```diff
@@ -401,7 +415,7 @@ convert.input <- function(input.id, outfolder, formatname, mimetype, site.id, st
       site.id <- input.args$newsite
     }
     
-    
+    if(exact.dates){allow.conflicting.dates = FALSE}
```
Comment (jsimkins2) [inline utils/R/convert.input.R]: if (exact.dates) {
     allow.conflicting.dates = FALSE
   }

File: utils/R/convert.input.R [hunk 2]
```diff
@@ -173,10 +187,10 @@ convert.input <- function(input.id, outfolder, formatname, mimetype, site.id, st
                    (end_date <= existing.input$end_date)) {
           
         
-            # There's an existing input that spans desired start/end dates. Use that one.
+         if (insert.new.file <- FALSE){ # There's an existing input that spans desired start/end dates. Use that one.
```
Comment (mdietze) [inline utils/R/convert.input.R]: Pretty sure you meant to use == here

File: utils/R/convert.input.R [hunk 3]
```diff
@@ -8,7 +8,7 @@
 
 convert.input <- function(input.id, outfolder, formatname, mimetype, site.id, start_date, 
                           end_date, pkg, fcn, con = con, host, browndog, write = TRUE, 
-                          format.vars, overwrite = FALSE, exact.dates = FALSE, ...) {
+                          format.vars, overwrite = FALSE, exact.dates = FALSE, allow.conflicting.dates = TRUE, ...) {
```
Comment (mdietze) [inline utils/R/convert.input.R]: I'm missing what allow.conflicting.dates actually means. By name, it seems to just be the opposite of exact.dates, but neither is documented

File: utils/R/convert.input.R [hunk 4]
```diff
@@ -402,18 +428,33 @@ convert.input <- function(input.id, outfolder, formatname, mimetype, site.id, st
...
+                                      in.prefix = result$dbfile.name[1], 
+                                      siteid = site.id, 
+                                      startdate = start_date,
+                                      nddate = end_date, 
+                                      mimetype, 
+                                      formatname, 
+                                      parentid = parent.id,
+                                      con = con, 
+                                      hostname = machine$hostname,
+                                      allow.conflicting.dates = allow.conflicting.dates)
```
Comment (serbinsh) [inline utils/R/convert.input.R]: @tonygardella I tried out the logic for conflicting dates as you have now (i was getting errors) and it seems to be working!

Comment (tonygardella) [inline utils/R/convert.input.R]: Cool!  but there's some cases where it doesn't still so I need to keep fixing and testing.

Comment (mdietze) [inline utils/R/convert.input.R]: If the current PR fixes some cases and doesn't break current cases I'd say we move ahead and then do another PR with the next fix. Important to keep this moving forward with AGU so close

File: db/man/dbfile.check.Rd [hunk 5]
```diff
@@ -2,10 +2,13 @@
 % Please edit documentation in R/dbfiles.R
 \name{dbfile.check}
 \alias{dbfile.check}
-\alias{dbfile.input.check}
-\alias{dbfile.posterior.check}
-\title{Check for a file in the input/dbfiles tables}
+\title{Check for a file in the dbfiles tables}
 \usage{
+<<<<<<< HEAD
```
Comment (robkooper) [inline db/man/dbfile.check.Rd]: remove conflict

File: modules/data.atmosphere/R/met.process.R [hunk 6]
```diff
@@ -137,7 +137,7 @@ met.process <- function(site, input_met, start_date, end_date, model,
   if (stage$download.raw) {
     raw.data.site.id <- ifelse(is.null(register$siteid), new.site$id, register$siteid)
     
-    raw.id <- PEcAn.data.atmosphere:::.download.raw.met.module(dir = dir,
+    raw.id <- .download.raw.met.module(dir = dir,
```
Comment (serbinsh) [inline modules/data.atmosphere/R/met.process.R]: @tonygardella did you mean to delete "PEcAn.data.atmosphere::" here?

Comment (tonygardella) [inline modules/data.atmosphere/R/met.process.R]: Yep I had added that accidentally as I was testing individual scripts. Originally it was  just ```.download.raw.met.module```.

File: db/R/dbfiles.R [hunk 7]
```diff
@@ -253,7 +254,7 @@ dbfile.posterior.insert <- function(filename, pft, mimetype, formatname, con, ho
 ##'
 ##' This will check the dbfiles, inputs, machines and formats tables to see if the
 ##' file exists
-##' @name dbfile.check
+##' @name dbfile.posterior.check
```
Comment (mdietze) [inline db/R/dbfiles.R]: name conflict still exists.

File: modules/data.atmosphere/R/met.process.R (hunk 6 above)
Comment (serbinsh) [inline modules/data.atmosphere/R/met.process.R]: Yeah I can see that now, thanks @mdietze.  Makes sense that it is in the lib so don't need to link to the lib

File: db/R/dbfiles.R (hunk 7 above)
Comment (tonygardella) [inline db/R/dbfiles.R]: The @name on Line 257 Goes with function on line 270 which is dbfile.posterior.check. There was a mismatch earlier that stuck ```dbfile.posterior.check``` and ```dbfile.input.insert``` along with ```dbfile.check``` in the same .Rd file. This changes it so that each function has their own .Rd file. If this is how it's supposed to be I can change it back.

Comment (mdietze) [inline db/R/dbfiles.R]: Sorry, you're OK. Didn't realize the amount of code Github was hiding from me between this line and the next chunk

File: utils/R/convert.input.R [hunk 8]
```diff
@@ -130,6 +134,9 @@ convert.input <- function(input.id, outfolder, formatname, mimetype, site.id, st
         
         existing.input <- db.query(paste0("SELECT * FROM inputs WHERE id=", existing.dbfile[["container_id"]]),con)
         
+
```
Comment (robkooper) [inline utils/R/convert.input.R]: should probably make sure we don't have excessive new lines here.
